cmd/gitserver/server/vcs_dependencies_syncer.go

```diff
@@ -85,15 +86,43 @@ func (s *vcsPackagesSyncer) CloneCommand(ctx context.Context, remoteURL *vcs.URL
 	}
 
 	// The Fetch method is responsible for cleaning up temporary directories.
-	if err := s.Fetch(ctx, remoteURL, GitDir(bareGitDirectory)); err != nil {
+	if err := s.Fetch(ctx, remoteURL, GitDir(bareGitDirectory), ""); err != nil {
 		return nil, errors.Wrapf(err, "failed to fetch repo for %s", remoteURL)
 	}
 
 	// no-op command to satisfy VCSSyncer interface, see docstring for more details.
 	return exec.CommandContext(ctx, "git", "--version"), nil
 }
 
-func (s *vcsPackagesSyncer) Fetch(ctx context.Context, remoteURL *vcs.URL, dir GitDir) (err error) {
+var versionAlreadySynced = errors.New("Version already exists")
```

Comment:
Would it make sense to include the version?

```suggestion
type versionAlreadySyncedErr struct {
  version string
}
func (e versionAlreadySyncedErr) Error() string { return fmt.Sprintf("version %s already synced", e.version) }
```

Reply:
I didn't include a version so that it's possible to test `== versionAlreadySynced`. I doubt the version itself is helpful for debugging because 1) this error is safe to silently ignore because it indicates everything is OK and 2) the call-site already has the version in case they want to log something

Reply:
Alright, fine with me!

But just FYI on this:
> I didn't include a version so that it's possible to test == versionAlreadySynced

You can use `errors.HasType(err, &versionAlreadySyncedErr{})`. See: https://docs.sourcegraph.com/dev/background-information/languages/go_errors#use-of-errors-hastype